### DISCUSSION TITLE
Julia Codegen Plugin

### DISCUSSION THREAD

**Discussion by markhalonen** [dotansimha/graphql-code-generator#6159] [Ideas]
Hello, I would like to build a codegen plugin for a Julia client. https://julialang.org/

Given a query/mutation, the plugin should generate the __minimum__ amount of code, not a type for the entire GraphQL schema.

I poked around and it seems like everyone generates the __entire__ graphQL schema to a file. I don't want to do this because my GraphQL API is massive and I'm only using a small part of it.

Are there any plugin examples that do this? I could not find any.

Thanks!

**Comment by dotansimha** [comment]
Thank you for creating it @markhalonen ! 
We are not highly familiar with Julia language, but we are really open for new ideas and concepts. So feel free to create a PR if you wish!
